blockchain/chain_sync/src/sync.rs

```diff
@@ -617,28 +631,36 @@ where
 
         let b = block.clone();
 
-        let db = Arc::clone(&self.chain_store.db);
         let parent_clone = parent_tipset.clone();
         // check messages to ensure valid state transitions
-        let x = task::spawn_blocking(move || Self::check_block_msgs(db, b, &parent_clone));
+        let sm = self.state_manager.clone();
+        let x = task::spawn_blocking(move || Self::check_block_msgs(sm, b, parent_clone));
```

Comment:
Did it turn out that spawn_blocking was correct? Seems weird because I would think it would make this syncronous (just confirming this was looked into)

Reply:
Yeah I talked to Eric and I think what we determined is that spawn_blocking since it is very CPU intensinve

Reply:
I talked to Ashanti this morning. It is correct. Spawn blocking isn't to make it sychronous or anything. It to spawn a thread from a dedicated pool of threads in the async std runtime. This dedicated pool of threads is for CPU intensive tasks (which in this case checking signatures can be quite heavy (hence blocking)).

Reply:
oops yeah, I had a lapse and was mixing up spawn_blocking with block_on